### PR TITLE
feat(vm): add vm modular support

### PR DESCRIPTION
**Pull request by jose-pablo-camacho**
xref: https://sysdig.atlassian.net/browse/SSPROD-46144

### CODE REVIEW

File: modules/agentless-scan/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# GCP Agentless Scanning Module
+
+This Module creates the resources required to scan hosts on Google Cloud Projects. Before applying the changes defined 
+in this module, the following operations need to be performed on the target GCP environment:
+
+- The APIs needed for the CDR/CIEM feature are listed below:
```
Comment (ravinadhruve10) [inline modules/agentless-scan/README.md:6]: Is this correct, as in do we need this?

File: modules/agentless-scan/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+| <a name="requirement_terraform"></a> [terraform](#requirement\_terraform) | >= 1.0    |
+| <a name="requirement_google"></a> [google](#requirement\_google)          | >= 4.21.0 |
+| <a name="requirement_sysdig"></a> [sysdig](#requirement\_sysdig)          | >= 1.34.0 |
+| <a name="requirement_random"></a> [random](#requirement\_random)          | >= 3.1    |
+
+## Modules
+
+No modules.
+
+## Resources
```
Comment (ravinadhruve10) [inline modules/agentless-scan/README.md:48]: could you please check and ensure we have all the resources (including sysdig component) from both main.tf and organizational.tf added here?

File: modules/agentless-scan/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+| <a name="input_suffix"></a> [suffix](#input\_suffix)                                                                 | Optional. Suffix word to enable multiple deployments with different naming<br/>(Workload Identity Pool and Providers have a soft deletion on Google Platform that will disallow name re-utilization)<br/>By default a random value will be autogenerated. | `string`       | `null`                      |    no    |
+
+## Outputs
+
+| Name                                                                                                         | Description                                                                                     |
+|--------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------|
+| <a name="agentless_scan_component_id"></a> [agentless\_scan\_component\_id](#agentless\_scan\_component\_id) | Component identifier of Agentless Scan integration created in Sysdig Backend for Log Ingestion  |
```
Comment (ravinadhruve10) [inline modules/agentless-scan/README.md:87]: could you please update the description, it mentions Log Ingestion.

File: modules/agentless-scan/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,5 @@
+output "agentless_scan_component_id" {
+  value       = "${sysdig_secure_cloud_auth_account_component.gcp_agentless_scan.type}/${sysdig_secure_cloud_auth_account_component.gcp_agentless_scan.instance}"
+  description = "Component identifier of Agentless Scan integration created in Sysdig Backend for Log Ingestion"
```
Comment (ravinadhruve10) [inline modules/agentless-scan/outputs.tf:3]: same here for component description

**Review by ravinadhruve10 — COMMENTED** [review]
Few comments.

File: modules/agentless-scan/README.md (hunk 1 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/README.md:6]: That was a typo, it should be:
`- The APIs needed for the VM feature are listed below:`

Comment (jose-pablo-camacho) [reply modules/agentless-scan/README.md:6]: just changed it

File: modules/agentless-scan/README.md (hunk 3 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/README.md:87]: done

File: modules/agentless-scan/outputs.tf (hunk 4 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/outputs.tf:3]: done

File: modules/agentless-scan/README.md (hunk 2 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/README.md:48]: done, added missing resource

File: modules/agentless-scan/variables.tf [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  default     = false
+}
+
+variable "organization_domain" {
+  type        = string
+  description = "(Optional) Organization domain. e.g. sysdig.com"
+  default     = ""
+}
+
+variable "role_name" {
```
Comment (haresh-suresh) [inline modules/agentless-scan/variables.tf:18]: Similar to CDR lets remove this variable and have the naming controlled with random suffixing within the module implementation

File: modules/agentless-scan/README.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+| <a name="input_organization_domain"></a> [organization\_domain](#input\_organization\_domain)                        | Optional. If `is_organizational=true` is set, its mandatory to specify this value, with the GCP Organization domain. e.g. sysdig.com                                                                                                                      | `string`       | `null`                      |    no    |
+| <a name="input_role_name"></a> [role\_name](#input\_role\_name)                                                      | Optional. Name for the Worker Role on the Customer infrastructure                                                                                                                                                                                         | `string`       | `"SysdigAgentlessHostRole"` |    no    |
```
Comment (haresh-suresh) [inline modules/agentless-scan/README.md:79]: Let's remove this as well

File: modules/agentless-scan/main.tf [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+  service_account_id = google_service_account.controller.name
+  role               = "roles/iam.workloadIdentityUser"
+  member             = "principalSet://iam.googleapis.com/${google_iam_workload_identity_pool.agentless.name}/attribute.sa_id/${var.sysdig_account_id}"
+}
+
+#-----------------------------------------------------------------------------------------
+# Custom IAM roles and bindings
+#-----------------------------------------------------------------------------------------
+
+resource "google_project_iam_custom_role" "controller" {
```
Comment (haresh-suresh) [inline modules/agentless-scan/main.tf:150]: nit: could we call this `controller_role` ?

File: modules/agentless-scan/main.tf [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+  count = var.is_organizational ? 0 : 1
+
+  project     = var.project_id
+  role_id     = "${var.role_name}Discovery${local.suffix}"
+  title       = "${var.role_name}, for Host Discovery"
+  permissions = local.host_discovery_permissions
+}
+
+resource "google_project_iam_binding" "controller_custom" {
```
Comment (haresh-suresh) [inline modules/agentless-scan/main.tf:159]: nit: could we call this `controller_binding` ?

File: modules/agentless-scan/organizational.tf [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+data "google_organization" "org" {
+  count  = var.is_organizational ? 1 : 0
+  domain = var.organization_domain
+}
+
+#-----------------------------------------------------------------------------------------
+# Custom IAM roles and bindings
+#-----------------------------------------------------------------------------------------
+
+resource "google_organization_iam_custom_role" "controller" {
```
Comment (haresh-suresh) [inline modules/agentless-scan/organizational.tf:14]: nit: let's call this `controller_role` too ?

File: modules/agentless-scan/organizational.tf [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  count = var.is_organizational ? 1 : 0
+
+  org_id      = data.google_organization.org[0].org_id
+  role_id     = "${var.role_name}Discovery${title(local.suffix)}"
```
Comment (haresh-suresh) [inline modules/agentless-scan/organizational.tf:18]: I don't see the point of titling a random suffix ? Let's get rid of it ?

File: modules/agentless-scan/organizational.tf [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  members = [
+    "serviceAccount:${google_service_account.controller.email}",
+  ]
+}
+
+resource "google_organization_iam_custom_role" "worker_role" {
+  count = var.is_organizational ? 1 : 0
+
+  org_id      = data.google_organization.org[0].org_id
+  role_id     = "${var.role_name}Scan${title(local.suffix)}"
```
Comment (haresh-suresh) [inline modules/agentless-scan/organizational.tf:37]: same here as https://github.com/sysdiglabs/terraform-google-secure/pull/39/files#r1762135039

File: modules/agentless-scan/main.tf [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+  instance   = "secure-scanning"
+  version    = "v0.1.0"
+  service_principal_metadata = jsonencode({
+    gcp = {
+      workload_identity_federation = {
+        pool_provider_id = data.sysdig_secure_agentless_scanning_assets.assets.gcp.worker_identity != null ? google_iam_workload_identity_pool_provider.agentless[0].name : var.sysdig_account_id != null ? google_iam_workload_identity_pool_provider.agentless_gcp[0].name : null
+      }
+      email = google_service_account.controller.email
+    }
+  })
```
Comment (haresh-suresh) [inline modules/agentless-scan/main.tf:207]: Do we need / require any explicit dependencies ?

File: modules/agentless-scan/variables.tf (hunk 5 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/variables.tf:18]: done

File: modules/agentless-scan/README.md (hunk 6 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/README.md:79]: done

File: modules/agentless-scan/main.tf (hunk 8 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/main.tf:159]: done

File: modules/agentless-scan/organizational.tf (hunk 9 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/organizational.tf:14]: done

File: modules/agentless-scan/main.tf (hunk 7 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/main.tf:150]: done

File: modules/agentless-scan/organizational.tf (hunk 10 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/organizational.tf:18]: yeah I just deleted all the `title()` functions, were part of previous uber install

File: modules/agentless-scan/organizational.tf (hunk 11 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/organizational.tf:37]: deleted it

File: modules/agentless-scan/main.tf (hunk 12 above)
Comment (jose-pablo-camacho) [reply modules/agentless-scan/main.tf:207]: not as of now, everything was working fine on tests

File: modules/config-posture/README.md [hunk 13]
```diff
@@ -16,7 +16,7 @@ If instrumenting an Organziation, the following resources will be created:
 - A cloud account component in the Sysdig Backend, associated with the GCP project and with the required component to serve the config posture functions.
 
 Note:
-- The outputs from the foundational module, such as `sysdig_secure_account_id` are needed as inputs to the other features/integrations modules for subsequent modular installs.
+- The outputs from the foundational module, such as `sysdig_secure_project_id` are needed as inputs to the other features/integrations modules for subsequent modular installs.
```
Comment (haresh-suresh) [inline modules/config-posture/README.md:19]: I believe this is an unintended change ? This should read `sysdig_secure_account_id`

Comment (jose-pablo-camacho) [reply modules/config-posture/README.md]: done

**Pull request merged**